pykickstart/commands/realm.py

```diff
@@ -101,3 +103,37 @@ def parse(self, args):
 
     def _getParser(self):
         return KSOptionParser(prog="realm", description="define an Active Directory realm to join", version=F19)
+
+
+class F42_Realm(F19_Realm):
+    removedKeywords = KickstartCommand.removedKeywords
+    removedAttrs = KickstartCommand.removedAttrs
+
+    def _parseJoin(self, args):
+        try:
+            # We only support these args
+            opts, remaining = getopt.getopt(args, "", ("client-software=",
```

Comment:
There may be a nicer way to do this than repeating the whole function. I'm undecided how much it matters at this point (but if there is ANOTHER change later on then I'd like to see a common function with args passed to it for each version). It's up to you if you want to try to make this more generic and share it between the F19 and F42 classes.

Reply:
Sounds sensible - I've had a go at this and also added 'os-name' and 'os-version' arguments should someone want to set them (see 0.16.3) https://gitlab.freedesktop.org/realmd/realmd/-/tags.

Reply:
I'll come back to this once I've worked out how to sufficiently solve https://github.com/pykickstart/pykickstart/pull/515#discussion_r1884360700